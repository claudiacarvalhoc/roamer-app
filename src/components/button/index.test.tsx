import React from 'react';
import Button, { ButtonProps } from './index';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const mockOnClick = jest.fn();
const getShallowWrapper = (props: ButtonProps) => shallow(<Button {...props} />);


describe('<Button />', () => {
    it('render correctly with primary type', () => {
        const props: ButtonProps = {
            text: 'Button name',
            type: 'primary',
            onClick: mockOnClick,
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('render correctly with secondary type', () => {
        const props = {
            type: 'secondary'
        }
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('should call onClick method when button is clicked', () => {
        const props: ButtonProps = {
            text: 'Button name',
            type: 'primary',
            onClick: mockOnClick,
        };
        const component = getShallowWrapper(props);
        component.simulate('click');
        expect(mockOnClick).toHaveBeenCalled();
    });
});
