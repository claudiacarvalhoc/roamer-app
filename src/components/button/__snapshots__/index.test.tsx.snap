// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render correctly with primary type 1`] = `
<button
  className="button primary"
  onClick={[MockFunction]}
>
  Button name
</button>
`;

exports[`<Button /> render correctly with secondary type 1`] = `
<button
  className="button secondary"
/>
`;
